src/main/java/core/basesyntax/SalaryInfo.java

```diff
@@ -41,6 +44,51 @@
      */
     public String getSalaryInfo(String[] names, String[] data, String dateFrom, String dateTo)
             throws Exception {
-        return null;
+        LocalDate fromData = convertationToData(dateFrom);
+        LocalDate toData = convertationToData(dateTo);
+        if (toData.isBefore(fromData) || toData.isEqual(fromData)) {
+            throw new IllegalDateParametersException("Wrong parameters");
+        }
+        payrollAndWriteInNames(names, data, fromData, toData);
+        StringBuilder report = new StringBuilder("Отчёт за период ");
+        report.append(dateFrom).append(" - ").append(dateTo).append("\n");
+        for (int i = 0; i < names.length; i++) {
+            report.append(names[i]);
+            if (i != names.length - 1) {
+                report.append("\n");
+            }
+        }
+        return report.toString();
+    }
+
+    private void payrollAndWriteInNames(String[] names, String[] data,
+                                        LocalDate fromData, LocalDate toData) {
+        for (int i = 0; i < names.length; i++) {
+            int sum = 0;
+            for (String worker : data) {
+                String[] splitWorker = worker.split(" ");
+                if (splitWorker[1].equals(names[i])) {
+                    if (checkData(fromData, toData, splitWorker[0])) {
+                        sum += Integer.parseInt(splitWorker[2])
+                                * Integer.parseInt(splitWorker[3]);
+                    }
+                }
+            }
+            if (sum != 0) {
+                names[i] += " - " + sum;
```

Comment:
bad idea, as you use String concatenation instead of StringBuilder. write this in a getSalaryInfo method, appending exactly to StringBuilder